### ISSUE TITLE
serviceResolved not triggered sometimes after serviceInfo.setText() on Android

### ISSUE THREAD

**Issue by DmitriyYakovlev** [jmdns/jmdns#129]
android application has next logic.
ServiceInfo has settings.
this is Hashtable<String, String> of some values.
normally, when some settings are changed, application will set new such hash table to 
serviceInfo.setText(settings)

if all works fine. serviceResolved will be triggered for ServiceListener
this works good, but sometimes this can be broken due to unknown reasons.

serviceInfo.setText(settings) works without exception
but serviceResolved does not triggers at all.

**Comment by DmitriyYakovlev** [comment]
More observations.
when problem happens all additional serviceInfo.setText() invocation does not works.
re-registering of service solves this and serviceResolved start triggering again

**Comment by DmitriyYakovlev** [comment]
Can you please, give me some advice how to solve this.

**Comment by ViToni** [comment]
Maybe you could explain your use-case a little more in detail.
Are you setting the values of ServiceInfo or do you expect that settings are changed when a new message arrives?

**Comment by DmitriyYakovlev** [comment]
Yes, of course. I'am ready to provide explanation.
1) Initial setting up of jmdns
When my app registers service it uses initial service info. Like below.
	jmdns.registerService(serviceInfo);

This service info created in next way.
				serviceInfo = ServiceInfo.create(
					SERVICE_INFO_TYPE,
					SERVICE_INFO_NAME,
					JmdnsConst.JMDNS_SERVICE_INFO_PORT,
					0, 0, true,
					settings);

It contains innitial settings.
This is Hashtable<String, String> settings.
it contains key-value pairs.
like
unique_id : 3434343434
ipv4 : 192.160.1.1
other : "some_text"

it contains 5 such pairs.

Every device with app in team registers in such manner.
Thus, every device can see such data for all other devices when triggers method:  
serviceResolved(ServiceEvent ev) of ServiceListener

2) When some device decides to change for example, field 
other : "some_text_changed"

it do
serviceInfo.setText(settings);
this settings is a 
Hashtable<String, String> settings,
but with new data.

and then, all other devices can see triggering of serviceResolved(ServiceEvent ev):
then, they can parse  ServiceEvent ev and grab changed data.

3) Problem.
This approach works great.
but, i have noticed during deep testing, that sometimes does not triggers at all
serviceResolved(ServiceEvent ev)

I mean, 
device A changes his serviceInfo.setText(settings);
and devices A, B, C has triggered serviceResolved(ServiceEvent ev) 
then, this works for some time.
but, sometimes I can see that 

device A changes his serviceInfo.setText(settings),
but serviceResolved is silent for A, B, C devices.

Unfortunately, this is influence other app functionality.
Please, tell me if you have more questions.

**Comment by ViToni** [comment]
Looking at the relevant code parts (of JmDNS) I would suggest you to add some logging to it.

This would improve the code and you could see why the updates don't occur every time.

**Comment by DmitriyYakovlev** [comment]
Do you mean, to import JmDNS as library project to my app instead of using of 
compile 'org.jmdns:jmdns:3.5.2'
and then populate JmDNS related code with additional logging ?

**Comment by ViToni** [comment]
I mean that you should enrich the existing JmDNS with logging and then use this version in your app so you see what happens.
Then you should create a PR out of it because IMHO JmDNS need some extra logging.

**Comment by ViToni** [comment]
One more aspect to your issue: this is UDP so it might be relevant if some messages don't get through.
Im might be not relevant if updates don't happen at all (or stop completely after some time...)

**Comment by DmitriyYakovlev** [comment]
Good point regarding UDP. I have noticed in my practice problems with UDP messages.
Thus, if problem is really with UDP this is impossible to fix. 
Only way to improve this is to notify client code in any way that problem is happens during sending.

**Comment by Abu-Abdullah** [comment]
i believe this case is related to #97  
the reason is that JmDNS stopped after sometime. it might not be a network issue. you can check this by analyzing the response from the JmDNS using tcpdump

`tcpdump -i wlan0 port 5353`
